### ISSUE TITLE
feature selection tuning parameter(s)

### ISSUE THREAD

**Issue by topepo** [business-science/gamsnip#9]
It would make sense to have a main argument for the `select` argument: 

> `select`: If this is `TRUE` then gam can add an extra penalty to each term so that it can be penalized to zero. This means that the smoothing parameter estimation that is part of fitting can completely remove terms from the model. If the corresponding smoothing parameter is estimated as zero then the extra penalty has no effect. Use `gamma` to increase level of penalization.

and

> `gamma`:  Increase this beyond 1 to produce smoother models. gamma multiplies the effective degrees of freedom in the GCV or UBRE/AIC. coden/gamma can be viewed as an effective sample size in the GCV score, and this also enables it to be used with REML/ML. Ignored with P-RE/ML or the efs optimizer.

Maybe use `select_features` for `select` and `adjust_deg_free` for `gamma`?

**Comment by AlbertoAlmuinha** [comment]
Thanks for this  @topepo . This is a good idea in my opinion. Also agree with the proposed names. For my part, let's do it.

**Comment by mdancho84** [comment]
I agree too! I actually wasn't aware that we can apply smooths to all of features. I do have a couple questions about factors since I'm wondering how the s() "by" argument works when automating but this is some research on my part.
